components/scripts/lib/paths.sh

```diff
@@ -46,9 +46,19 @@ relative_path() {
     printf '%s\n' "$result"
 }
 
+pushd_with_project_dir() {
```

Comment:
I suggest renaming this to `pushd_to_project_dir` since we're changing directories to the project_dir (it's similar to something like `cd_to_project_dir`).